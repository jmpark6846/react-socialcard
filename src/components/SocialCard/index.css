
.card-wrapper{
	margin-top: 10px;
}
.social-card{
	width:492px;
}

.social-card:hover{
    box-shadow: 0 .5rem 1rem rgba(0,0,0,.15)!important;
}

.social-card img{
	height:257px;
}

.social-card .card-text{
	font-size: 13px;
	color: darkslategray;
}

.card-body {
    padding: 10px 15px;
}

.card-title {
    margin-bottom: 7px;
}

.card-content{
	color:black;
}
