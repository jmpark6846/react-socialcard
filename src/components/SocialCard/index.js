import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const SocialCard = ({metaTags}) => 
  <div className="card-wrapper">
    <div className="card social-card shadow-sm" >
      <a href={metaTags.url} target="_blank">
        <div className="card-content">
          <img className="card-img-top" src={metaTags.image.url} alt="Card image cap" />
          <div className="card-body">
            <h5 className="card-title">{metaTags.title}</h5>
            <p className="card-text">{metaTags.description} </p>
          </div>
        </div>  
      </a>
    </div>
  </div>

SocialCard.propTypes = {
  metaTags: PropTypes.object
}

export default SocialCard;
