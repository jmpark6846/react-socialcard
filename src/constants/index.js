const BASE_URL = "http://opengraph.io/api/1.1/site/";
const TARGET_URL = "https://www.robinwieruch.de/learn-react-before-using-redux/";
const APP_ID = "?app_id=5ae0777b9b03547407c63e6d";
//TODO : 환경 변수에 따라 APP_ID 설정

const metaTagData = {
	description: "Facts about React that should be known before using Redux (or MobX). Most important: Learn React first, then opt-in Redux but only if you need it. There are various techniques in React.js for scaling your local state management. Only if these are not sufficient, use a library like Redux ...",
	image: {
		url:"https://www.robinwieruch.de/img/posts/learn-react-before-using-redux/banner_640.jpg",
	},
	title: "8 things to learn in React before using Redux - RWieruch",
	type: "website",
	url: "https://www.robinwieruch.de/learn-react-before-using-redux/",
}

export { BASE_URL, TARGET_URL, APP_ID, metaTagData };
