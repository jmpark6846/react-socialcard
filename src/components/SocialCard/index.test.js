import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import SocialCard from '.';
import { metaTagData } from '../../constants';
Enzyme.configure({ adapter: new Adapter() });
const props = {
		metaTags: {
			description: "Facts about React that should be known before using Redux (or MobX). Most important: Learn React first, then opt-in Redux but only if you need it. There are various techniques in React.js for scaling your local state management. Only if these are not sufficient, use a library like Redux ...",
			image: {
				url:"https://www.robinwieruch.de/img/posts/learn-react-before-using-redux/banner_640.jpg",
			},
			title: "8 things to learn in React before using Redux - RWieruch",
			type: "website",
			url: "https://www.robinwieruch.de/learn-react-before-using-redux/",
		}
	}

describe('SocialCard', ()=>{
	
	it('renders without crashing', () => {
	  const div = document.createElement('div');

	  ReactDOM.render(<SocialCard {...props}/>, div);
	  ReactDOM.unmountComponentAtNode(div);
	});

	test('has a valid snapshot', () => {
		const component = renderer.create(<SocialCard {...props}/>);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('shows socialcard', ()=>{
		const element = shallow(
			<SocialCard {...props}/>
		);
		expect(element.find('.card-wrapper').length).toBe(1);
	});
});

