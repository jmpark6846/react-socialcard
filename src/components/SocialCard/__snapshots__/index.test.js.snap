// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialCard has a valid snapshot 1`] = `
<div
  className="card-wrapper"
>
  <div
    className="card social-card shadow-sm"
  >
    <a
      href="https://www.robinwieruch.de/learn-react-before-using-redux/"
      target="_blank"
    >
      <div
        className="card-content"
      >
        <img
          alt="Card image cap"
          className="card-img-top"
          src="https://www.robinwieruch.de/img/posts/learn-react-before-using-redux/banner_640.jpg"
        />
        <div
          className="card-body"
        >
          <h5
            className="card-title"
          >
            8 things to learn in React before using Redux - RWieruch
          </h5>
          <p
            className="card-text"
          >
            Facts about React that should be known before using Redux (or MobX). Most important: Learn React first, then opt-in Redux but only if you need it. There are various techniques in React.js for scaling your local state management. Only if these are not sufficient, use a library like Redux ...
             
          </p>
        </div>
      </div>
    </a>
  </div>
</div>
`;
